Core/src/game/functions/ints/count/sizeBiggestGroup/CountSizeBiggestGroup.java

```diff
@@ -0,0 +1,348 @@
+							
+							int[] locnUp = new int[0];
+							int[] indexUp = new int[0];
+							
+							if (isVisibleFn != null && isVisibleFn.eval(context) == true) { // ced
+								final List<game.util.graph.Step> stepsbis = context.game().board().topology().trajectories() // ced
+										.steps(SiteType.Vertex, to, SiteType.Vertex, AbsoluteDirection.Upward); // ced
+								
+								for (final Step stepbis : stepsbis) // ced
+								{
+									final int toSite = stepbis.to().id(); // ced
+									if (cs.what(toSite, type) != 0) // ced
+										indexUpwards.add(toSite); // ced
+								}
+								
+								locnUp = locnUpwards.toArray();	// ced
+								indexUp = indexUpwards.toArray();	// ced
+								
+//								System.out.println("locA: " + Arrays.toString(locnUp));
+//								System.out.println("indexA: " + Arrays.toString(indexUp));
+							}
+
+							// If we already have it we continue to look the others.
+							if (groupSitesBS.get(to))
+								continue;
+							
+//							System.out.println("-------------");
+//							System.out.println("From: " + site);
+//							System.out.println("To: " + to);
+//							System.out.println("inter: " + getIntersectionLength(locnUp, indexUp));
+
+							context.setTo(to);
+							if (condition.eval(context) && !(getIntersectionLength(locnUp, indexUp) >= 2) && !((to == 8 && cs.what(9, type) != 0) || (to == 10 && cs.what(11, type) != 0) || (to == 14 && cs.what(15, type) != 0) || (to == 18 && cs.what(19, type) != 0) || (to == 20 && cs.what(21, type) != 0)))
+							{
+//								System.out.println(to);
+								groupSites.add(to);
+								groupSitesBS.set(to);
+							}
+						}
+					}
+
+					++i;
+				}
+
+				if (groupSites.size() > count)
+					count = groupSites.size();
+
+				sitesChecked.or(groupSitesBS);
+			}
+		}
+
+		context.setTo(origTo);
+
+		return count;
+	}
+	
+	//-------------------------------------------------------------------------
+	
+		/**
+	     * function to compute length of intersection of two int arrays
+	     */
+		private static int getIntersectionLength(int[] array1, int[] array2) {
+	        Set<Integer> set1 = new HashSet<>();
+	        Set<Integer> intersection = new HashSet<>();
+	        
+	        // Add elements of the first array to the set
+	        for (int num : array1) {
+	            set1.add(num);
+	        }
+	        
+	        // Check elements of the second array against the set
+	        for (int num : array2) {
+	            if (set1.contains(num)) {
+	                intersection.add(num);
+	            }
+	        }
+	        
+	        return intersection.size();
+	    }
+		
+	//-------------------------------------------------------------------------
+
+	@Override
+	public boolean isStatic()
+	{
+		return false;
+	}
+
+	@Override
+	public String toString()
+	{
+		return "Groups()";
+	}
+
+	@Override
+	public long gameFlags(final Game game)
+	{
+		long gameFlags = condition.gameFlags(game);
+		gameFlags |= SiteType.gameFlags(type);
```

Comment:
this function and some of the ones below (concepts, all the evalContext ones, ...) should operate on all member variables of the class, not just `type`